crates/blockifier/src/execution/native/syscall_handler.rs

```diff
@@ -807,11 +807,29 @@ impl<'state> StarknetSyscallHandler for &mut NativeSyscallHandler<'state> {
 
     fn sha256_process_block(
         &mut self,
-        _prev_state: &[u32; 8],
-        _current_block: &[u32; 16],
-        _remaining_gas: &mut u128,
+        prev_state: &[u32; 8],
+        current_block: &[u32; 16],
+        remaining_gas: &mut u128,
     ) -> SyscallResult<[u32; 8]> {
-        todo!()
+        const SHA256_STATE_SIZE: usize = 8;
+
+        self.substract_syscall_gas_cost(
+            remaining_gas,
+            self.execution_context.gas_costs().sha256_process_block_gas_cost,
+        )?;
+
+        let data_as_bytes: sha2::digest::generic_array::GenericArray<
+            u8,
+            sha2::digest::typenum::U64,
+        > = sha2::digest::generic_array::GenericArray::from_exact_iter(
```

Comment:
(Nitpick) I think there is no need to declare this super long type, type inference works in this case. Is there any specific reason of why you declared it?

Reply:
I've removed the explicit type.